app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectInformationActivity.kt

```diff
@@ -108,30 +113,65 @@ class ProjectInformationActivity : AppCompatActivity() {
         }
     }
 
+    private fun onApplyClick(
+        applyButton: Button,
+        candidatureDatabase: CandidatureDatabase,
+        projectId: ProjectId,
+    ) {
+        if (alreadyApplied) {
+            candidatureDatabase.removeCandidature(
+                projectId,
+                userId!!,
+                {
+                    showToast(getString(R.string.success), Toast.LENGTH_SHORT)
+                    alreadyApplied = !alreadyApplied
+                    setApplyButtonText(applyButton, alreadyApplied)
+                },
+                { showToast(getString(R.string.failure), Toast.LENGTH_SHORT) }
+            )
+        } else {
+            candidatureDatabase.pushCandidature(
+                projectId,
+                userId!!,
```

Comment:
You are not handling the null pointer exceptions from the !! operator. You could simply check that the Id is not null inside your function if you are worried about it.

Reply:
this function is called only in an if (userId != null)